plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -743,6 +827,32 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
             if p in kwargs:
                 del kwargs[p]
 
+    # Clean up the cloud-init variables
+
+    if 'ipconfig' in kwargs:
+        # 'break down' the ipconfig dictionary into a series of the following
+        # ipconfig0: ip=192.168.1.2/24,gw=192.168.1.1
+        # ipconfig1: ip=192.168.2.2/24
+        for ipconfig in kwargs['ipconfig']:
+            try:
+                label = 'ipconfig{0}'.format(ipconfig)
+                config = 'ip={0}'.format(
+                    kwargs['ipconfig'][ipconfig]['ip'])
+                if 'gw' in ipconfig:
+                    config += ',gw={0}'.format(
+                        kwargs['ipconfig'][ipconfig]['gw'])
+                kwargs[label] = config
+            except Exception as e:
+                module.fail_json(
+                    msg='ipconfig cannot convert {0} to an integer: {1}'.format(
+                        ipconfig, e),
+                    data=kwargs['ipconfig'][ipconfig])
+        del kwargs['ipconfig']
+    # 'sshkeys' param expects an urlencoded string
+    if 'sshkeys' in kwargs:
+        urlencoded_ssh_keys = quote(kwargs['sshkeys'], safe='')
+        kwargs['sshkeys'] = str(urlencoded_ssh_keys)
```

Comment:
Isn't this already handled by proxmoxer and requests ?

Reply:
![Screenshot from 2020-11-08 09-01-30](https://user-images.githubusercontent.com/600106/98460049-0575f580-21a1-11eb-9162-ac92d8060395.png)

Nope.